### PR TITLE
Prevent swiping while scrolling vertically

### PR DESCRIPTION
**Pull request by axeldelafosse**
Hey guys!

Thank you for this carousel, it's exactly what I was looking for. However we experienced a undesirable behaviour on mobile: while scrolling vertically the carousel is swiping.

This is a fix for this bug. Happy to help.

Cheers

### CODE REVIEW

**Comment by mrbinky3000** [comment]
Hey, is there someplace where I can see the buggy behavior.  A couple of us here were playing around with our carousel on our phones: android and iphone, and yes, it is possible to scroll move the carousel perpendicular to the direction you are scrolling, but we're not sure it's a bug.  I just want to make sure I'm seeing the same thing you're seeing.

Check out this page on our site...
https://www.express.com/clothing/women/original-fit-long-sleeve-print-portofino-shirt/pro/09701223/color/FLORAL%20PRINT

... and let me know if it's happening there.
(future readers a month or so from now, our inventory changes all the time, so just try a different product)

So, to recap: 
A) send me a link to your project if you can.
B) Check out the link I sent you and tell me if it's happening there.

- Matt

**Comment by axeldelafosse** [comment]
Hey Matt,

Yes it's the same bug!

Unfortunately, I can't send you a direct link to our webapp. To see the carousel, you need to be logged in to our webapp via our Messenger bot...

I can send you a before/after screencast if you want. Did you test the fix?

Let me know

**Comment by mrbinky3000** [comment]
Hmm.  In my opinion, I don't know if this is a bug.  But I'm all for making the carousel accomodate user needs.  SO,  I think this could be a feature.  Like a new prop on the CarouselProvider:  lockWhenPageScrolling.  This will lock scrolling perpendicular to the page scrolling.

Do you mind if I make that new feature and give you credit?

**Comment by axeldelafosse** [comment]
Yes no problem, I was thinking about doing that too.

Thank you!

**Comment by axeldelafosse** [comment]
Hello Matt. Do you need help? I can do it if you want.

**Comment by triniMahn** [comment]
@axeldelafosse, we are experiencing the same issue, except in the reverse scenario: when scrolling horizontally through the carousel, if the swipe direction isn't aligned exactly with the x-axis then the page scrolls (or appears to "bounce").

Testing with the link you previously posted yields the aforementioned result as well.
Here are some links to our beta application where we're also experiencing this issue: 
- Page scrolling: http://listen-beta-qa.azurewebsites.net/podcasts
- "bouncing" behaviour : http://listen-beta-qa.azurewebsites.net/live

Do you think your proposed "lockWhenPageScrolling" setting will resolve our issue as well?

Also, let me know how we can help?

Thank you!

**Comment by mrbinky3000** [comment]
@axeldelafosse @triniMahn Sorry for the delay.  Black Friday -> Xmas is a busy time in retail.  I've been swamped with work stuff.  However, I have some time coming up where I can get that feature added for @axeldelafosse  and fix the issue for @triniMahn.

**Comment by mrbinky3000** [comment]
@axeldelafosse v1.10.0 now has a property on CarouselProvider called lockOnWindowScroll().  I incorporated your work with some minor changes to avoid a warning about trying to update a component that was already unmounted.   Thank you for suggesting the new feature!   Let me know if it does what you want.

**Comment by axeldelafosse** [comment]
Thanks ☺️

**Comment by aniruddhashevle** [comment]
@mrbinky3000 : Even after adding lockOnWindowScroll, on a mobile browser, this issue is still there. I'm using version `1.27.1`.

**Comment by michaelmarconi** [comment]
@aniruddhashevle — did you manage to cure this problem in the end?
@mrbinky3000 — I'm also seeing the issue on mobile (Safari).  It is possible to scroll the carousel by dragging diagonally, although it gets a bit jerky, so it seems something is kicking in.  Could this be a regression?
